Tasks/Common/MSBuildHelpers/ArgumentFunctions.ps1

```diff
@@ -24,9 +24,31 @@ function Format-MSBuildArguments {
         if ($MaximumCpuCount) {
             $MSBuildArguments = "$MSBuildArguments /m"
         }
+        
+        $userAgent = Get-UserAgentString
+        if ($userAgent) {
+            $MSBuildArguments = "$MSBuildArguments /p:_MSDeployUserAgent=`"$userAgent`""
+        }
 
         $MSBuildArguments
     } finally {
         Trace-VstsLeavingInvocation $MyInvocation
     }
 }
+
+function Get-UserAgentString
+{
+    [CmdletBinding()]
+	param()
+
+	Trace-VstsEnteringInvocation $MyInvocation
+    try {
+          if ($env:AZURE_HTTP_USER_AGENT) {
```

Comment:
This environment variable is now added when we createplan in distributed task sdk. In function setsystemvariable